spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/JdbcAggregateTemplateIntegrationTests.java

```diff
@@ -492,8 +493,194 @@ public void saveAndLoadLongChainWithoutIds() {
+				c3.chain2.add(c2);
+
+				for (int _1 = 0; _1 <= 1; _1++) {
+
+					NoIdListChain1 c1 = new NoIdListChain1();
+					c1.oneValue = c2.twoValue + _1;
+					c2.chain1.add(c1);
+
+					for (int _0 = 0; _0 <= 1; _0++) {
+
+						NoIdListChain0 c0 = new NoIdListChain0();
+						c0.zeroValue = c1.oneValue + _0;
+						c1.chain0.add(c0);
+
+					}
+				}
+			}
+
+		}
+
+		NoIdListChain4 saved = template.save(chain4);
+
+		assertThat(saved.four).isNotNull();
+
+		NoIdListChain4 reloaded = template.findById(saved.four, NoIdListChain4.class);
+
+		SoftAssertions.assertSoftly(softly -> {
+
+			softly.assertThat(reloaded.fourValue).isEqualTo("v");
+			softly.assertThat(reloaded.chain3).hasSize(2);
+			for (int _3 = 0; _3 <= 1; _3++) {
+
+				NoIdListChain3 c3 = reloaded.chain3.get(_3);
+				softly.assertThat(c3.threeValue).isEqualTo(chain4.fourValue + _3);
+				softly.assertThat(c3.chain2).hasSize(2);
+
+				for (int _2 = 0; _2 <= 1; _2++) {
+
+					NoIdListChain2 c2 = c3.chain2.get(_2);
+					softly.assertThat(c2.twoValue).isEqualTo(c3.threeValue + _2);
+					softly.assertThat(c2.chain1).hasSize(2);
+
+					for (int _1 = 0; _1 <= 1; _1++) {
+
+						NoIdListChain1 c1 = c2.chain1.get(_1);
+						softly.assertThat(c1.oneValue).isEqualTo(c2.twoValue + _1);
+						softly.assertThat(c1.chain0).hasSize(2);
+
+						for (int _0 = 0; _0 <= 1; _0++) {
+
+							NoIdListChain0 c0 = c1.chain0.get(_0);
+							softly.assertThat(c0.zeroValue).isEqualTo(c1.oneValue + _0);
+
+						}
+					}
+				}
+
+			}
+
+		});
+
+		template.deleteById(saved.four, NoIdListChain4.class);
+
+		SoftAssertions.assertSoftly(softly -> {
+
+			softly.assertThat(count("NO_ID_LIST_CHAIN4")).isEqualTo(0);
+			softly.assertThat(count("NO_ID_LIST_CHAIN3")).isEqualTo(0);
+			softly.assertThat(count("NO_ID_LIST_CHAIN2")).isEqualTo(0);
+			softly.assertThat(count("NO_ID_LIST_CHAIN1")).isEqualTo(0);
+			softly.assertThat(count("NO_ID_LIST_CHAIN0")).isEqualTo(0);
+		});
+
+	}
+
+	@Test // DATAJDBC-223
+	public void saveAndLoadLongChainOfMapsWithoutIds() {
+
+		NoIdMapChain4 chain4 = new NoIdMapChain4();
+		chain4.fourValue = "v";
+		for (int _3 = 0; _3 <= 1; _3++) {
+
+			NoIdMapChain3 c3 = new NoIdMapChain3();
+			c3.threeValue = chain4.fourValue + _3;
+			chain4.chain3.put(asString(_3), c3);
+
+			for (int _2 = 0; _2 <= 1; _2++) {
+
+				NoIdMapChain2 c2 = new NoIdMapChain2();
+				c2.twoValue = c3.threeValue + _2;
+				c3.chain2.put(asString(_2),c2);
+
+				for (int _1 = 0; _1 <= 1; _1++) {
+
+					NoIdMapChain1 c1 = new NoIdMapChain1();
+					c1.oneValue = c2.twoValue + _1;
+					c2.chain1.put(asString(_1),c1);
+
+					for (int _0 = 0; _0 <= 1; _0++) {
+
+						NoIdMapChain0 c0 = new NoIdMapChain0();
+						c0.zeroValue = c1.oneValue + _0;
+						c1.chain0.put(asString(_0),c0);
+
+					}
+				}
+			}
+
+		}
+
+		NoIdMapChain4 saved = template.save(chain4);
+
+		assertThat(saved.four).isNotNull();
+
+		NoIdMapChain4 reloaded = template.findById(saved.four, NoIdMapChain4.class);
+
```

Comment:
Same goes for assertions: Wrapping assertions into a function closure increases nesting level and due to the length it is hard to understand what happens here.